sys/include/net/conn/udp.h

```diff
@@ -51,6 +51,9 @@ typedef struct conn_udp conn_udp_t;
  * @param[in] family    The family of @p addr (see @ref net_af).
  * @param[in] port      The local UDP port for @p conn.
  *
+ * @note    @ref conn_udp_recvfrom needs to be called from the same thread as
+ *          for this function.
+ *
```

Comment:
> Futhermore, I couldn't find documentation explicitly for gnrc_conn_udp.

there is none. It's just the implementation of this API for GNRC and it *should* not have different behavior than documented here ;-)